frame/support/src/event.rs

```diff
@@ -498,31 +455,14 @@ macro_rules! impl_outer_event {
 
 #[macro_export]
 #[doc(hidden)]
-macro_rules! __impl_outer_event_json_metadata {
+macro_rules! __impl_outer_event_metadata {
 	(
 		$runtime:ident;
 		$event_name:ident;
 		$( $module_name:ident::Event < $( $generic_params:path ),* > $( $instance:ident )?, )*;
 	) => {
 		impl $runtime {
-			#[allow(dead_code)]
-			pub fn outer_event_metadata() -> $crate::event::OuterEventMetadata {
```

Comment:
this function is the main goal of `_impl_outer_event_json_metadata`, to implement the metadata of the outer event.
But I think nobody uses it anymore and use construct_runtime instead.

I actually think we can remove `impl_outer_event`, or maybe we can continue to implement this function here, I'll make a PR to remove these `impl_outer_*`.

Reply:
Ah yes I see `impl_outer_event!` is only used in tests now

Reply:
I opened https://github.com/paritytech/substrate/pull/9307